operator/pkg/helmreconciler/wait.go

```diff
@@ -162,10 +158,12 @@ func waitForResources(objects object.K8sObjects, cs kubernetes.Interface, l *pro
 		}
 	}
 	dr, dnr := deploymentsReady(deployments)
+	dsr, dsnr := daemonsetsReady(daemonsets)
+	stsr, stsnr := statefulsetsReady(statefulsets)
 	nsr, nnr := namespacesReady(namespaces)
 	pr, pnr := podsReady(pods)
-	isReady := dr && nsr && pr
-	notReady := append(append(nnr, dnr...), pnr...)
+	isReady := dr && nsr && dsr && stsr && pr
+	notReady := append(append(append(append(nnr, dnr...), pnr...), dsnr...), stsnr...)
```

Comment:
nit: This looks a little confusing, even though this is functional, can this be simplified?